### PR TITLE
Testing/app v1

### PR DESCRIPTION
**Pull request by attom2**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Testing

## What is the new behavior?

- Adds async testing for app.js and refactors get4RdmPokemon to give 4 unique numbers

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React, { useEffect, useState } from 'react';
 import './App.css';
-import Header from '../Header/Header'
-import Game from '../Game/Game'
-import {getAllPokemon} from '../../ApiCalls'
+import Header from '../Header/Header';
+import Game from '../Game/Game';
+import {getAllPokemon} from '../../ApiCalls';
```
Comment (attom2) [inline src/Components/App/App.js:5]: Setting up mock functions

File: src/Components/App/App.js [hunk 2]
```diff
@@ -24,9 +24,14 @@ const App = () => {
 
   const get4RdmPokemon = () => {
     const pokemons = [{}, {}, {}, {}]
-    return pokemons.map(emptySlot => {
-      const randomIndex = Math.floor(Math.random() * allPokemon.length)
-      return allPokemon[randomIndex]
+    let randomIndexes = []
```
Comment (attom2) [inline src/Components/App/App.js:27]: Refactor to not have duplicate pokemon displaying to button choices

File: src/Components/App/App.test.js [hunk 3]
```diff
@@ -1,9 +1,113 @@
...
+import { singlePokemonMockData} from './Pokemon-Mock-Data';
+import { act } from 'react-dom/test-utils';
+// import '@testing-library/jest-dom/extend-expect';
+
+
+jest.mock("../../ApiCalls");
+
+describe('App', () => {
+
+  getAllPokemon.mockResolvedValue( [
```
Comment (attom2) [inline src/Components/App/App.test.js:17]: Mocking out fetch calls

File: src/Components/App/App.test.js [hunk 4]
```diff
@@ -1,9 +1,113 @@
...
+      {
+        "name": "charmander",
+        "url": "https://pokeapi.co/api/v2/pokemon/4/"
+      },
+      {
+        "name": "charmeleon",
+        "url": "https://pokeapi.co/api/v2/pokemon/5/"
+  }])
+
+  getSinglePokemon.mockResolvedValue(singlePokemonMockData)
```
Comment (attom2) [inline src/Components/App/App.test.js:39]: Lives in another file, can easily just mock the 5 things we look at (height, weight, etc)

File: src/Components/App/App.test.js [hunk 5]
```diff
@@ -1,9 +1,113 @@
...
+        <App />
+      </MemoryRouter>
+    );
+    const gameNav = getByText('Game')
+    fireEvent.click(gameNav);
+    const winningStreak = await waitFor(() => getByRole('heading', {name: "Winning streak: 0"}));
+    debug();
+    expect(winningStreak).toBeInTheDocument();
+
+    //Still need to check if winning button (charmander) is clicked, should add 1 to win counter
```
Comment (attom2) [inline src/Components/App/App.test.js:77]: More tests needed for game functionality, also sad paths

File: src/Components/Game/Game.js [hunk 6]
```diff
@@ -3,11 +3,11 @@ import './Game.css'
...
 import {getSinglePokemon} from '../../ApiCalls'
 
-const Game = ({pokemons, get4RdmPokemon}) => {
+const Game = ({ get4RdmPokemon}) => {
   const [singlePokemon, setSinglePokemon] = useState({});
   const [winCounter, setWinCounter] = useState(0);
-  const [pokemonChoices, setPokemonChoices] = useState(pokemons)
-
+  const pokemons = get4RdmPokemon();
+  const [pokemonChoices, setPokemonChoices] = useState(pokemons);
```
Comment (emontealvo) [inline src/Components/Game/Game.js:10]: I like this refactoring to a single parameter, my vote is to keep it. Single concern would be some conflicts/failing test from Corbin's branch in the future.

File: src/Components/App/App.js (hunk 2 above)
Comment (emontealvo) [reply src/Components/App/App.js:27]: Nice attention to detail!

Comment (MarchCorbin) [reply src/Components/App/App.js:27]: Good Idea to implement the bug fix!

**Pull request merged**